src/components/SignUpForm/index.test.js

```diff
@@ -0,0 +1,409 @@
+        value: "name",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /lastName/i }), {
+      target: {
+        value: "surname",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: "passcode",
+      },
+    });
+
+    fireEvent.click(screen.getByRole("checkbox"));
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /firstName/i }).value).toBe(
+      "name"
+    );
+    expect(screen.getByRole("textbox", { name: /lastName/i }).value).toBe(
+      "surname"
+    );
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@mail.com"
+    );
+    expect(screen.getByPlaceholderText(/^password/i).value).toBe("password");
+    expect(screen.getByPlaceholderText(/repeat password/i).value).toBe(
+      "passcode"
+    );
+    expect(screen.getByRole("checkbox").checked).toBe(true);
+  });
+
+  it("should display accept terms error when not accept terms", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
+      target: {
+        value: "name",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /lastName/i }), {
+      target: {
+        value: "surname",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /firstName/i }).value).toBe(
+      "name"
+    );
+    expect(screen.getByRole("textbox", { name: /lastName/i }).value).toBe(
+      "surname"
+    );
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@mail.com"
+    );
+    expect(screen.getByPlaceholderText(/^password/i).value).toBe("password");
+    expect(screen.getByPlaceholderText(/repeat password/i).value).toBe(
+      "password"
+    );
+    expect(screen.getByRole("checkbox").checked).toBe(false);
+  });
+
+  it("should not display error when value is valid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
```

Comment:
in general I'd say these tests are too long and helper functions need to be used liberally for both readability and maintainability 

you should think about the ease of updating these tests if something changes. right now there's a lot of scenarios where a small change requires updating in like 10 places in the test file

Reply:
When I wrote that code, I did think that it will be too long. I think this task is not so easy and need much time, so I added a TODO comment to work on it later if you do not mind.